docs/errors/E0290.md

```diff
@@ -0,0 +1,29 @@
+# E0290: using '===' or '!==' against an array literal always produces the same result
```

Comment:
Yeah, it definitely looks better this way, but when I was fixing this, I was thinking if it could be a good idea to addionally give a warning like `this comparison seems unintentional` when the user uses a non-strict operator. I know that in some cases `x == []` might be true, but I don't think that there's a valid reason to write that (maybe there is, I'm not extremely familiar with js)


So we could have three diagnostics for arrays:
For strict operators (=== and !==):
  - `'{0} []' is always {1}` ('=== []' is always true; '!== []' is always false)
  -  `using '{0}' against an array literal does not compare items`

for non-strict operators (== and !=):
  - `this comparison seems unintentional` 

Do you think it could be a good idea?

Reply:
Let's not warn on `==` or `!=` for now. That can be a different task.